src/main/kotlin/org/rust/lang/core/psi/ext/RsMacroCall.kt

```diff
@@ -53,6 +54,21 @@ val RsMacroCall.expansion: List<ExpansionResult>?
         expandMacro(this)
     }
 
+fun RsMacroCall.processExpansionRecursively(processor: (ExpansionResult) -> Boolean): Boolean =
+    processExpansionRecursively(processor, 0)
+
+private fun RsMacroCall.processExpansionRecursively(processor: (ExpansionResult) -> Boolean, recursionDepth: Int): Boolean {
```

Comment:
Same here: we can improve this function by using `any` method of the list, like this:
```kotlin
private fun RsMacroCall.processExpansionRecursively(processor: (ExpansionResult) -> Boolean, recursionDepth: Int): Boolean {
    if (recursionDepth > DEFAULT_RECURSION_LIMIT) return true

    return expansion.orEmpty().any { it.processRecursively(processor, recursionDepth) }
}

private fun ExpansionResult.processRecursively(processor: (ExpansionResult) -> Boolean, recursionDepth: Int) = when (this) {
    is RsMacroCall -> this.processExpansionRecursively(processor, recursionDepth + 1)
    else -> processor(this)
}
```

This way there would be only one return point and it is easier to read (imo)